proposals/partial-type-inference.md

```diff
@@ -0,0 +1,738 @@
+# Partial type inference
+
+> Note: This proposal was created because of championed [Partial type inference](https://github.com/dotnet/csharplang/issues/1349). It is a continuation of the proposed second version published in [csharplang/discussions/7467](https://github.com/dotnet/csharplang/discussions/7467) and the first version published in [csharplang/discussions/7286](https://github.com/dotnet/csharplang/discussions/7286)
+
+## Summary
```

Comment:
Chatted with @jaredpar about the whole proposal. There's three parts that I can see:
1. allowing free/floating type arguments in method invocation (`M<int, _>(...)`)
2. allowing free/floating type arguments in object creation **with floating arity** (`new C<>(...)`)
3. allowing object initializers contribute to the inference in object creation (`new C<>(...) { ...initializer... }`)

The second and third part seem much more complicated as they introduce bidirectional type inference.

What we would propose instead as core design proposal is:
1. allowing free/floating type arguments in method invocation [KEEP]
2. allowing free/floating type arguments in object creation **with fixed arity** using the same syntax: `new C<_, _>()`
The benefit of restricting object creation to fixed/known arity is that the flow for object creation remains more like today: pick a single type and apply overload resolution to its constructors (and overload resolution can now deal with fixed vs. floating type arguments).

Those two core parts would be very useful, straightforward and not very controversial with LDM.

We would recommend either removing or clearly separating out (potentially in a separate doc) the two parts that are controversial and complex:
1. allowing floating arity in object creation
2. allowing initializers to contribute to type inference in object creation

Reply:
Should by fixed by commit [ea87f2f](https://github.com/dotnet/csharplang/pull/7582/commits/ea87f2f973aff156f9467c56d4e1861e3c7306bb). I moved the mentioned parts (initializers and floating arity) to alternatives.